src/states/elder/mod.rs

```diff
@@ -665,178 +656,151 @@ impl Elder {
     fn handle_message_signature(
         &mut self,
         msg: AccumulatingMessage,
-        pub_id: PublicId,
-    ) -> Result<(), RoutingError> {
-        if !self.chain.is_peer_elder(&pub_id) {
+        src: PublicId,
+        _outbox: &mut dyn EventBox,
+    ) -> Result<Transition> {
+        if !self.chain.is_peer_elder(&src) {
             debug!(
                 "{} - Received message signature from invalid peer {}, {:?}",
-                self, pub_id, msg
+                self, src, msg
             );
             return Err(RoutingError::InvalidSource);
         }
 
-        if let Some(signed_msg) = self.sig_accumulator.add_proof(msg) {
-            let signed_msg = HopMessageWithBytes::new(signed_msg, &self.log_ident())?;
-            self.handle_signed_message(signed_msg)?;
+        if let Some(msg) = self.sig_accumulator.add_proof(msg) {
+            self.handle_accumulated_message(msg)?
         }
-        Ok(())
+
+        Ok(Transition::Stay)
     }
 
-    // If the message is for us, verify it then, handle the enclosed routing message and swarm it
-    // to the rest of our section when destination is targeting multiple; if not, forward it.
-    fn handle_signed_message(&mut self, msg: HopMessageWithBytes) -> Result<(), RoutingError> {
-        if !self.routing_msg_filter.filter_incoming(&msg).is_new() {
+    fn handle_accumulated_message(&mut self, msg: Message) -> Result<()> {
+        // FIXME: this is almost the same as `Base::handle_new_deserialized_message` - find a way
```

Comment:
Might be worth doing as part of that PR now it is smaller.

Reply:
Wondering if `add_proof` should return a `MessageWithBytes`.

Reply:
That fix me is still there (slightly changed). Probably best to fix it with an additional commit.

Reply:
The FIXME is there because it's not addressed yet and likely won't be in this PR. To address it properly would require some changes which I think are out of scope for this PR.

Reply:
What change would it require?

it seem to me that the only difference is that we want to do something slightly different if we did the accumulation.
 1 - either pass a parameter to allow the different behavior.
 2 - is there a good reason for us doing different thing in this case?

But I'm happy for it to be in another PR.

Reply:
> What change would it require?

The `handle_accumulated_message` function is also called from `send_routing_message` which 1) doesn't take `outbox` and 2) doesn't return `Transition`. That means that `handle_accumulate_message` also can't take/return those. But `handle_new_deserialized_message` does take/return them.

My preferred solution to this would be to get rid of the `outbox` argument altogether and use a member variable instead (that would simplify the code overall, not just in this place). That would still leave us with the `Transition` return type but I think we can ignore that here because `Elders` never transition on receiving a message. Anyway, I think that is something for a future PR.

Reply:
ok